nomenclature/codelist.py

```diff
@@ -364,6 +335,35 @@ def read_excel(cls, name, source, sheet_name, col, attrs=None):
 
         return cls(name=name, mapping=mapp)
 
+    def check_illegal_characters(
+        self, config: NomenclatureConfig = None
+    ) -> Dict[str, Code]:
+        """Check that no illegal characters are left in codes after tag replacement"""
+        forbidden = ["{", "}"]
+        if config and config.illegal_characters:
+            forbidden += config.illegal_characters
+
+        def _check_string(value):
+            if isinstance(value, str):
+                if any(char in value for char in forbidden):
+                    raise ValueError(
+                        f"Unexpected character in {self.name}: '{code.name}'."
+                        " Check for illegal characters and/or if tags were spelled correctly."
+                    )
+            elif isinstance(value, dict):
+                for v in value.values():
```

Comment:
I know you copied this over but now that I look at it, I'm thinking we might want to validate both keys and values of a dictionary.